znai-core/src/main/java/com/twosigma/znai/extensions/file/CodeReferencesTrait.java

```diff
@@ -18,13 +15,13 @@
  *
  * sort of like a plugin trait. Maybe a good idea to formalize traits on plugins interface level.
  */
-public class CodeReferences {
+public class CodeReferencesTrait {
```

Comment:
If you want to make it Trait, could you extract the part to become an interface?

Reply:
I plan to do a bigger refactoring around it. Renaming is a first step. I am not ready to formally introduce a trait yet, need to think about capabilities and interface first.

Reply:
Then can you just role back the name change, since the class is not Trait.

Reply:
I renamed to avoid confusion with newly introduced class `DocReferences`. They sound similar but perform different functions. One is data holder and another acts like a trait. 
I plan to add another trait like class for handling highlighting logic. At that point I was planning to start formalizing it more.